app/common/view.c

```diff
@@ -231,14 +253,61 @@ zxerr_t h_review_update_data() {
         viewdata.pageIdx = 0;
         return zxerr_ok;
     }
+
+    if (h_paging_intro_screen()) {
+        char* intro_key = NULL;
+        char* intro_value = NULL;
+
+    #if defined(REVIEW_SCREEN_ENABLED) && defined(APP_BLIND_MODE_ENABLED)
+        switch (viewdata.itemIdx) {
+            case 0:
+                intro_key = PIC(review_key);
+                if(mustReply == REVIEW_ADDRESS) {
+                    intro_value = PIC(review_addrvalue);
+                } else if (mustReply == REVIEW_UI) {
+                    intro_value = PIC(review_configvalue);
+                } else {
+                    intro_value = PIC(review_txvalue);
+                }
+                break;
+            case 1:
+                intro_key = PIC(blindsigning_key);
+                intro_value = PIC(blindsigning_value);
+                break;
+            default:
+                return zxerr_no_data;
+        }
+    #elif defined(REVIEW_SCREEN_ENABLED)
+        intro_key = PIC(review_key);
+        if(mustReply == REVIEW_ADDRESS) {
+            intro_value = PIC(review_addrvalue);
+        } else if (mustReply == REVIEW_UI) {
+            intro_value = PIC(review_configvalue);
+        } else {
+            intro_value = PIC(review_txvalue);
+        }
+    #elif defined(APP_BLIND_MODE_ENABLED)
+        intro_key = PIC(blindsigning_key);
+        intro_value = PIC(blindsigning_value);
+    #else
+        return zxerr_no_data;
+    #endif
```

Comment:
I find this a confusing. It is difficult to understand what is doing. This can lead to bugs. The logic should be more explicit